File: src/Problem017.go
package main

import (
	"fmt"
	"time"
)

func spell(n int) string {
	if n == 1 {
		return "one"
	}
	if n == 2 {
		return "two"
	}
	if n == 3 {
		return "three"
	}
	if n == 4 {
		return "four"
	}
	if n == 5 {
		return "five"
	}
	if n == 6 {
		return "six"
	}
	if n == 7 {
		return "seven"
	}
	if n == 8 {
		return "eight"
	}
	if n == 9 {
		return "nine"
	}
	if n == 10 {
		return "ten"
	}
	if n == 11 {
		return "eleven"
	}
	if n == 12 {
		return "twelve"
	}
	if n == 13 {
		return "thirteen"
	}
	if n == 14 {
		return "fourteen"
	}
	if n == 15 {
		return "fifteen"
	}
	if n == 16 {
		return "sixteen"
	}
	if n == 17 {
		return "seventeen"
	}
	if n == 18 {
		return "eighteen"
	}
	if n == 19 {
		return "nineteen"
	}

	if n > 19 && n < 100 {
		tens := n / 10
		ones := n % 10
		if tens == 2 {
			return "twenty " + spell(ones)
		}
		if tens == 3 {
			return "thirty " + spell(ones)
		}
		if tens == 4 {
			return "forty " + spell(ones) //not spelled fourty!
		}
		if tens == 5 {
			return "fifty " + spell(ones)
		}
		if tens == 6 {
			return "sixty " + spell(ones)
		}
		if tens == 7 {
			return "seventy " + spell(ones)
		}
		if tens == 8 {
			return "eighty " + spell(ones)
		}
		if tens == 9 {
			return "ninety " + spell(ones)
		}
	}

	if n > 99 && n < 1000 {
		ending := n % 100
		hundreds := n / 100

		var endSpell string
		if ending == 0 {
			endSpell = ""
		} else {
			endSpell = " and " + spell(ending)
		}

		return spell(hundreds) + " hundred" + endSpell

	}

	if n == 1000 {
		return "one thousand"
	}

	return ""

}

func count(word string) int {
	if word == "" {
		return 0
	}
	if string(word[0]) == " " {
		return count(word[1:])
	}
	return 1 + count(word[1:])
}

func main() {
	starttime := time.Now()

	sum := 0
	for i := 1; i <= 1000; i++ {
		sum += count(spell(i))
	}

	fmt.Println(sum)

	fmt.Println("Elapsed time:", time.Since(starttime))
}

File: python/Problem017.py
def spell(number):
    if number < 20:
        small = {0:'', 1:'one', 2:"two", 3:"three", 4:"four", 5:"five",6:"six",7: "seven", 8:"eight",  9: "nine",  10: "ten",  11 : "eleven",  12 : "twelve",  13:  "thirteen" , 14 : "fourteen", 15 : "fifteen" ,  16:  "sixteen" ,  17 : "seventeen",  18 : "eighteen",  19 : "nineteen" }
        return small[number]

    if number > 19 and number < 100:
        tens = number / 10
        ones = number % 10

        big = {2:"twenty ", 3:"thirty ", 4:"forty ", 5:"fifty ", 6:"sixty ", 7:"seventy ", 8:"eighty ", 9:"ninety "}
        return big[tens]+spell(ones)

    if number > 99 and number < 1000:
        ending = number % 100
        hundreds = number / 100

        if ending == 0:
            return spell(hundreds) + "hundred"
        else:
            return spell(hundreds) + " hundred and " + spell(ending)

    if number == 1000:
        return "one thousand"


def count(word):
    if len(word) == 0:
        return 0
    if word[0] == " ":
        return count(word[1:])
    return 1 + count(word[1:])


total = 0
for x in xrange(1,1001):
    # print spell(x)
    total += count(spell(x))

print total